tests/e2e/improved_csi_idempotency.go

```diff
@@ -0,0 +1,837 @@
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, diskSize, nil, "", true, "")
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating PVCs using the Storage Class")
+		count := 0
+		fmt.Printf("VOLUME_OPS_SCALE is set to %v", volumeOpsScale)
+		for count < volumeOpsScale {
+			fmt.Printf("Creating %v pvc ", count)
+			pvclaims[count], err = fpv.CreatePVC(client, namespace,
+				getPersistentVolumeClaimSpecWithStorageClass(namespace, diskSize, storageclass, nil, ""))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			count++
+		}
+		pvclaims = append(pvclaims, pvclaim)
+
+		ginkgo.By("Waiting for all claims to be in bound state")
+		persistentvolumes, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			for _, claim := range pvclaims {
+				err := fpv.DeletePersistentVolumeClaim(client, claim.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			ginkgo.By("Verify PVs, volumes are deleted from CNS")
+			for _, pv := range persistentvolumes {
+				err := fpv.WaitForPersistentVolumeDeleted(client, pv.Name, framework.Poll, framework.PodDeleteTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from kubernetes",
+						pv.Spec.CSI.VolumeHandle))
+			}
+		}()
+
+		// Modify PVC spec to trigger volume expansion
+		// We expand the PVC while no pod is using it to ensure offline expansion
+		ginkgo.By("Expanding current pvc")
+		for _, claim := range pvclaims {
+			currentPvcSize := claim.Spec.Resources.Requests[v1.ResourceStorage]
+			newSize := currentPvcSize.DeepCopy()
+			newSize.Add(resource.MustParse("4Gi"))
+			framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
+			claims, err := expandPVCSize(claim, newSize, client)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(claims).NotTo(gomega.BeNil())
+		}
+
+		/* ginkgo.By("Restart CSI driver")
+		framework.Logf("Stopping CSI driver")
+		err = updateDeploymentReplicawithWait(client, 0, vSphereCSIControllerPodNamePrefix, csiSystemNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("Starting CSI driver")
+		err = updateDeploymentReplicawithWait(client, 1, vSphereCSIControllerPodNamePrefix, csiSystemNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintln("Sleeping for 5 buffer min"))
+		time.Sleep(pollTimeout) */
+
+		ginkgo.By("Waiting for controller volume resize to finish")
```

Comment:
Resize wont work untill a POD is created

Reply:
Please refer "Verify Offline volume expansion when VSAN-health is down" and
 "Verify online volume expansion when VSAN-health is down"